applications/FluidDynamicsApplication/custom_strategies/strategies/compressible_navier_stokes_explicit_solving_strategy_bfecc.h

```diff
@@ -0,0 +1,385 @@
+        BaseType::InitializeBFECCBackwardSubstep();
+        this->InitializeEverySubstep();
+    };
+
+    /**
+     * @brief Initialize the BFECC final substep
+     * This method is intended to implement all the operations required before each BFECC final substep
+     */
+    void InitializeBFECCFinalSubstep() override
+    {
+        BaseType::InitializeBFECCFinalSubstep();
+        PopDiffusiveConstants();
+        this->InitializeEverySubstep();
+    };
+
+    /**
+     * @brief Initialize the BFECC forward substep
+     * This method is intended to implement all the operations required before each BFECC initial forward substep
+     */
+    void FinalizeBFECCForwardSubstep() override
+    {
+        BaseType::FinalizeBFECCForwardSubstep();
+        this->FinalizeEverySubstep();
+    };
+
+    /**
+     * @brief Finalize the BFECC backward substep
+     * This method is intended to implement all the operations required before each BFECC backward substep
+     */
+    void FinalizeBFECCBackwardSubstep() override
+    {
+        BaseType::FinalizeBFECCBackwardSubstep();
+        this->FinalizeEverySubstep();
+    };
+
+    /**
+     * @brief Finalize the BFECC final substep
+     * This method is intended to implement all the operations required before each BFECC final substep
+     */
+    void FinalizeBFECCFinalSubstep() override
+    {
+        BaseType::FinalizeBFECCFinalSubstep();
+        PopDiffusiveConstants();
+        this->FinalizeEverySubstep();
+    };
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    struct Stash {
+        double conductivity = 0.0;
+        double dynamic_viscosity = 0.0;
+    } mDiffusionStash;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    void StashDiffusiveConstants()
+    {
+        auto& model_part = BaseType::GetModelPart();
+        if(model_part.NumberOfElements() == 0) return;
+
+        auto& properties = model_part.ElementsBegin()->GetProperties();
+
+        auto& r_conductivity = properties.GetValue(CONDUCTIVITY);
+        auto& r_dynamic_viscosity = properties.GetValue(DYNAMIC_VISCOSITY);
+
+        mDiffusionStash.conductivity = r_conductivity;
+        mDiffusionStash.dynamic_viscosity = r_dynamic_viscosity;
+
+        r_conductivity = 0;
+        r_dynamic_viscosity = 0;
+    }
+
+    void PopDiffusiveConstants()
+    {
+        auto& model_part = BaseType::GetModelPart();
+        if(model_part.NumberOfElements() == 0) return;
+
+        auto& properties = model_part.ElementsBegin()->GetProperties();
+
+        properties.SetValue(CONDUCTIVITY, mDiffusionStash.conductivity);
+        properties.SetValue(DYNAMIC_VISCOSITY, mDiffusionStash.dynamic_viscosity);
```

Comment:
Note that by doing this you will setting a zero value if `CONDUCTIVITY` (or `DYNAMIC_VISCOSITY`) were not present in the properties container before. I think this shouldn't be done. You only reset the previously existing values.

Reply:
Agreed. I implemented a very reduced version of C++17's std::optional to do this. When the transition is done, we can remove my struct and everything should work.